src/components/input/index.jsx

```diff
@@ -84,11 +85,17 @@ class Input extends React.Component {
         this.setValue(e.currentTarget.value)
     }
 
+    click(e) {
```

Comment:
Hmm I think this situation might be solved if we have the following UX:

Clicking on the input-dropdown should trigger the dropdown to be expanded and the input to be focused.

Blurring on the input-downdown should collapse the dropdown and blur the input.

So at any point clicking the input would mean clicking the input-dropdown which would always trigger the list to expand.